### ISSUE TITLE
Shoryuken Hangs

### ISSUE THREAD

**Issue by brianp** [phstc/shoryuken#118]
Hey,

I wasn't going to create an issue but I noticed this was an issue previously and the fix was to version lock celluloid at 0.15.2 (https://github.com/phstc/shoryuken/pull/11/files).

If there is a discussion group that may be a better place then the issue tracker.

Running:
ruby 2.1.5p273 (2014-11-13 revision 48405)
shoryuken (~> 1.0.2)
celluloid (0.16.0)

The Shoryuken process starts up fine. After about 30 minutes the process begins to hang. After about 50 minutes the process may die but is certainly no longer polling for messages.

After the process dies, if I try to restart the process manually (un-daemonized) with or without -v it simply hangs with no output.

I tracked memory usage and didn't notice any abnormalities.

Any specific debugging steps I can go through?

Cheers,

**Comment by phstc** [comment]
hey @brianp 

Could you try [TTIN](https://github.com/phstc/shoryuken/wiki/Signals#ttin) to check Shoryuken status?

Could you detail what your workers do and you Shoryuken configuration?

**Comment by brianp** [comment]
Hey,

TTIN reports 18 threads with 4 different backtraces. I've listed the 4 below. Interesting though that the process is supposed to start with only concurrency = 5.

The workers don't do anything heavy. Receive messages, write 1-3 rows to a table, potentially send of more SQS messages to another queue.

We startup shoryuken with the command:
```bash
bundle exec ./bin/shoryuken -r ./config/boot.rb -d -l /dev/null -c 5
```
 - The worker logs to syslog as defined in the app, so we pipe the forced log file on startup to /dev/null
 - boot.rb requires our entire GrapeFramework application, which loads the initializer and worker.

Shoryuken initializer:
```ruby
class Shoryuken::ErrorLogger
  def call(worker_instance, queue, sqs_msg, body)
    begin
      yield
    rescue StandardError => e
      NewRelic::Agent.notice_error(e)
      raise e
    end
  end
end

Shoryuken.configure_server do |config|
  require 'app/workers/sqs_poller'

  Shoryuken::Logging.logger = App.logger
  url = Settings.aws.inbound_queues.pds_notification_sqs_url

  config.options[:aws][:region] = URI(url).host.split('.')[1]
  config.queues << url.split('/').last

  config.server_middleware do |chain|
    chain.add Shoryuken::ErrorLogger
  end
end
```

Stack traces for threads:
```ruby
Jul 21 17:26:32 lp-page-data-service-fe602535 lp-page-data-service[31300]:
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/cli.rb:182:in `backtrace'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/cli.rb:182:in `block in handle_signal'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/cli.rb:179:in `each'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/cli.rb:179:in `handle_signal'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/cli.rb:53:in `run'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/bin/shoryuken:6:in `<top (required)>'\n./bin/shoryuken:16:in `load'\n./bin/shoryuken:16:in `<main>'

Jul 20 20:39:08 lp-page-data-service-d63f411d lp-page-data-service[19064]:
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/mailbox.rb:62:in `sleep'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/mailbox.rb:62:in `wait'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/mailbox.rb:62:in `block in check'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/timers-4.0.1/lib/timers/wait.rb:14:in `for'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/mailbox.rb:57:in `check'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/actor.rb:153:in `block in run'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/timers-4.0.1/lib/timers/group.rb:52:in `wait'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/actor.rb:150:in `run'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/actor.rb:130:in `block in start'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/thread_handle.rb:13:in `block in initialize'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/actor_system.rb:32:in `block in get_thread'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/internal_pool.rb:130:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/internal_pool.rb:130:in `block in create'

Jul 21 00:05:05 lp-page-data-service-fe602535 lp-page-data-service[1746]:
/usr/lib/ruby/2.1.5/lib/ruby/2.1.0/monitor.rb:185:in `lock'
/usr/lib/ruby/2.1.5/lib/ruby/2.1.0/monitor.rb:185:in `mon_enter'
/usr/lib/ruby/2.1.5/lib/ruby/2.1.0/monitor.rb:209:in `mon_synchronize'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:271:in `active_connection?'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract/connection_pool.rb:293:in `with_connection'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/config/initializers/closing_active_record_connections.rb:6:in `connection'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:262:in `execute_simple_calculation'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:224:in `perform_calculation'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:119:in `calculate'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:34:in `count'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation.rb:251:in `empty?'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation.rb:261:in `any?'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/app/service_objects/message_processor.rb:45:in `is_duplicate_message'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/app/service_objects/message_processor.rb:10:in `process'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/app/workers/sqs_poller.rb:11:in `perform'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/processor.rb:26:in `block (2 levels) in process'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:94:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:94:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/config/initializers/shoryuken.rb:4:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/active_record.rb:6:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/auto_delete.rb:6:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/exponential_backoff_retry.rb:9:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/timing.rb:14:in `block in call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/logging.rb:21:in `with_context'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/timing.rb:8:in `call'\
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:99:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:99:in `invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/processor.rb:25:in `block in process'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `public_send'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `dispatch'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:63:in `dispatch'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/future.rb:14:in `block in new'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/thread_handle.rb:13:in `block in initialize'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/actor_system.rb:32:in `block in get_thread'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/internal_pool.rb:130:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/internal_pool.rb:130:in `block in create'

Jul 21 17:17:40 lp-page-data-service-fe602535 lp-page-data-service[31300]:
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:303:in `query'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:303:in `block in execute'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_adapter.rb:373:in `block in log'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activesupport-4.1.4/lib/active_support/notifications/instrumenter.rb:20:in `instrument'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_adapter.rb:367:in `log'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:303:in `execute'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/mysql2_adapter.rb:228:in `execute'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:310:in `execute_and_free'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:451:in `columns'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/schema_cache.rb:93:in `block in prepare_default_proc'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/schema_cache.rb:44:in `yield'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/connection_adapters/schema_cache.rb:44:in `columns'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/model_schema.rb:214:in `columns'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/model_schema.rb:260:in `column_names'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:229:in `aggregate_column'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:252:in `execute_simple_calculation'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:224:in `perform_calculation'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:119:in `calculate'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation/calculations.rb:34:in `count'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation.rb:251:in `empty?'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/activerecord-4.1.4/lib/active_record/relation.rb:261:in `any?'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/app/service_objects/message_processor.rb:45:in `is_duplicate_message'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/app/service_objects/message_processor.rb:10:in `process'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/app/workers/sqs_poller.rb:11:in `perform'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/processor.rb:26:in `block (2 levels) in process'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:94:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:94:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/config/initializers/shoryuken.rb:4:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/active_record.rb:6:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/auto_delete.rb:6:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/exponential_backoff_retry.rb:9:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/timing.rb:14:in `block in call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/logging.rb:21:in `with_context'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/server/timing.rb:8:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:96:in `block in invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:99:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/middleware/chain.rb:99:in `invoke'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/shoryuken-1.0.2/lib/shoryuken/processor.rb:25:in `block in process'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `call'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `public_send'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `dispatch'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/calls.rb:63:in `dispatch'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle/ruby/2.1.0/gems/celluloid-0.16.0/lib/celluloid/future.rb:14:in `block in new'
/opt/lp-page-data-service/releases/lp-page-data-service-a87b748/vendor/bundle
```

**Comment by brianp** [comment]
Hey Pablo,

I've finally diagnosed this as not a Shoryuken problem. It came down to a mysql connection pooling issue. It just was never a problem until we introduced Shoryuken. Shoryuken exposed it for us when failed messages hit a specific retry threshold. It would retry constantly opening up new connections that were left hanging eventually causing the pool to dry up.

I'll close this as it's not a real issue.

Thanks for taking your time!
